src/parser/parser_flow.ml

```diff
@@ -1938,7 +1938,7 @@ end = struct
     and identifier_or_reserved_keyword env =
       let { lex_token; lex_value; lex_loc; _ } = lookahead env in
       match lex_token with
-      | T_IDENTIFIER -> Parse.identifier env, None
+      | _ when Peek.identifier env -> Parse.identifier env, None
```

Comment:
Hmm, maybe this should be

```JavaScript
| T_IDENTIFIER -> Parse.identifier env, None
| _ when Peek.identifier env ->          
    (lex_loc, Identifier.({
      name = lex_value;
      typeAnnotation = None;
      optional = false;
    })), None
```

Since I think `identifier_or_reserved_keyword` doesn't want to error here.